src/App.tsx

```diff
@@ -1,14 +1,52 @@
-import React from 'react';
+import React, { useMemo, useRef, useState } from 'react';
 import './App.scss';
 import { peopleFromServer } from './data/people';
+import { Person } from './types/Person';
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [isInput, setIsInput] = useState('');
+  const [isInputFocused, setInputFocused] = useState(false);
+  const [isAppliedInput, setIsAppliedInput] = useState('');
+  const [isSelectedPerson, setIsSelectedPerson] = useState(peopleFromServer[0]);
+  const [isSelected, setIsSelected] = useState(false);
+  const { name, born, died } = isSelectedPerson;
+
+  const timerId = useRef(0);
+
+  const filteredPeople = useMemo(() => {
+    return peopleFromServer.filter(person => person
+      .name.toLowerCase().includes(isAppliedInput.toLowerCase()));
+  }, [isAppliedInput]);
+
+  const handleInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setIsInput(event.target.value);
+
+    if (isAppliedInput !== event.target.value) {
+      window.clearTimeout(timerId.current);
+
+      timerId.current = window.setTimeout(() => {
+        setIsAppliedInput(event.target.value);
+      }, 500);
```

Comment:
let's move this delay logic to some separate function (debounce)

Reply:
Why do this if it is only used once?